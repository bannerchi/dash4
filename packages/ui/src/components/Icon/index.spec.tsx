/// <reference types="@types/jest" />
import { render } from '@testing-library/react';
import React from 'react';
import { Icon } from './index';

describe('Icon', () => {
	test('exist', () => {
		expect(typeof Icon).toBe('object');
	});
	test('render default', () => {
		const { container } = render(<Icon name="favorite" />);
		expect(container.firstChild).toMatchSnapshot();
	});
	test('render with className', () => {
		const { container } = render(<Icon className="test" name="favorite" />);
		expect(container.firstChild).toMatchSnapshot();
	});
	test('render in size s', () => {
		const { container } = render(<Icon size="s" name="favorite" />);
		expect(container.firstChild).toMatchSnapshot();
	});
	test('render in size m', () => {
		const { container } = render(<Icon size="m" name="favorite" />);
		expect(container.firstChild).toMatchSnapshot();
	});
	test('render in size l', () => {
		const { container } = render(<Icon size="l" name="favorite" />);
		expect(container.firstChild).toMatchSnapshot();
	});
	test('render in dark color', () => {
		const { container } = render(<Icon color="dark" name="favorite" />);
		expect(container.firstChild).toMatchSnapshot();
	});
	test('render in bright color', () => {
		const { container } = render(<Icon color="bright" name="favorite" />);
		expect(container.firstChild).toMatchSnapshot();
	});
	test('render with animation rotation-clockwise', () => {
		const { container } = render(<Icon animation="rotation-clockwise" name="favorite" />);
		expect(container.firstChild).toMatchSnapshot();
	});
	test('render with animation rotation-counter-clockwise', () => {
		const { container } = render(<Icon animation="rotation-counter-clockwise" name="favorite" />);
		expect(container.firstChild).toMatchSnapshot();
	});
	test('render with alignment center-in-content', () => {
		const { container } = render(<Icon align="center-in-content" name="favorite" />);
		expect(container.firstChild).toMatchSnapshot();
	});
});
