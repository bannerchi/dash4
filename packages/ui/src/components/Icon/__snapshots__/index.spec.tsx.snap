// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon render default 1`] = `
<span
  class="undefined Component-root-0-1-1"
>
  <i
    class="material-icons"
  >
    favorite
  </i>
</span>
`;

exports[`Icon render in bright color 1`] = `
<span
  class="undefined Component-root-0-1-1 Component-color-bright-0-1-9"
>
  <i
    class="material-icons"
  >
    favorite
  </i>
</span>
`;

exports[`Icon render in dark color 1`] = `
<span
  class="undefined Component-root-0-1-1 Component-color-dark-0-1-8"
>
  <i
    class="material-icons"
  >
    favorite
  </i>
</span>
`;

exports[`Icon render in size l 1`] = `
<span
  class="undefined Component-root-0-1-1 Component-size-l-0-1-4"
>
  <i
    class="material-icons Component-size-i-l-0-1-7"
  >
    favorite
  </i>
</span>
`;

exports[`Icon render in size m 1`] = `
<span
  class="undefined Component-root-0-1-1 Component-size-m-0-1-3"
>
  <i
    class="material-icons Component-size-i-m-0-1-6"
  >
    favorite
  </i>
</span>
`;

exports[`Icon render in size s 1`] = `
<span
  class="undefined Component-root-0-1-1 Component-size-s-0-1-2"
>
  <i
    class="material-icons Component-size-i-s-0-1-5"
  >
    favorite
  </i>
</span>
`;

exports[`Icon render with alignment center-in-content 1`] = `
<span
  class="undefined Component-root-0-1-1 Component-align-center-in-content-0-1-14"
>
  <i
    class="material-icons"
  >
    favorite
  </i>
</span>
`;

exports[`Icon render with animation rotation-clockwise 1`] = `
<span
  class="undefined Component-root-0-1-1"
>
  <i
    class="material-icons Component-anim-rotation-clockwise-0-1-11"
  >
    favorite
  </i>
</span>
`;

exports[`Icon render with animation rotation-counter-clockwise 1`] = `
<span
  class="undefined Component-root-0-1-1"
>
  <i
    class="material-icons Component-anim-rotation-counter-clockwise-0-1-13"
  >
    favorite
  </i>
</span>
`;

exports[`Icon render with className 1`] = `
<span
  class="test Component-root-0-1-1"
>
  <i
    class="material-icons"
  >
    favorite
  </i>
</span>
`;
